### PR TITLE
Make DDD work with Forte.

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
Some changes to make Forte and Psi compatible with each other again. Heavily coordinated with Lori.

Also gets rid of an unused function that was raising a compile-time warning.

## Checklist
- [x] Psi tests pass
- [x] Forte tests pass

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/driver/p4util/procutil.py [hunk 1]
```diff
@@ -516,6 +515,35 @@ def prepare_options_for_set_options():
...
+        except RuntimeError:
+            # If we're in this case, Forte hasn't pushed its options to Psi.
+            pass
+        else:
+            # Load changed Forte options into `flat_options`
+            for name, metadata in forte_options.dict().items():
+                if metadata["value"] != metadata["default_value"]:
+                    flat_options[f"forte__{name.lower()}"] = metadata["value"]
+            # Restore current module
+        finally:
```
Comment (loriab) [inline psi4/driver/p4util/procutil.py:545]:
```suggestion
        finally:
            # Restore current module
```

File: psi4/driver/schema_wrapper.py [hunk 2]
```diff
@@ -576,6 +576,23 @@ def run_json_qcschema(json_data, clean, json_serialization, keep_wfn=False):
...
+    else:
+        # Initialization tasks with Psi options.
+        psi_options = core.get_options()
+        current_module = psi_options.get_current_module()
+        # Get the current Forte options from Forte
+        forte_options = forte.ForteOptions()
+        forte.register_forte_options(forte_options)
+        psi_options.set_current_module('FORTE')
+        forte_options.push_options_to_psi4(psi_options)
+        # Restore current options
```
Comment (loriab) [inline psi4/driver/schema_wrapper.py:593]:
```suggestion
        # Restore current module
```

**Review by loriab — APPROVED** [review]
Once more the Options dragon has been mollified. Hooray and thanks!

Code suggestions only worth it if other commits needed.

File: psi4/src/psi4/liboptions/liboptions.h [hunk 3]
```diff
@@ -456,7 +456,6 @@ class PSI_API Options {
     std::vector<int> get_int_vector(std::string key) const;
     double* get_double_array(std::string key) const;
     std::vector<double> get_double_vector(std::string key) const;
-    const char* get_cstr(std::string key) const;
```
Comment (fevangelista) [inline psi4/src/psi4/liboptions/liboptions.h:459]: ++ for removing a function returning `char*`!

**Review by fevangelista — APPROVED** [review]

**Review by maxscheurer — APPROVED** [review]

**Pull request merged**